### PR TITLE
[Cherry-Pick] Move helper methods onto ContentHost/Capsule/Satellite

### PR DESCRIPTION
**Pull request by mshriver**
Backport of #8872

### CODE REVIEW

File: robottelo/hosts.py [hunk 1]
```diff
@@ -237,21 +266,39 @@ def install_katello_host_tools(self):
...
 
-    def install_katello_ca(self, sat_hostname=None):
-        """Downloads and installs katello-ca rpm on the broker virtual machine.
+    def install_katello_ca(self, satellite):
+        """Downloads and installs katello-ca rpm on the content host.
 
-        Uses common helper `install_katello_ca(hostname=None)`, but passes
-        `self.hostname` instead of the hostname as we are using fake hostnames
-        for broker virtual machines.
+        :param str satellite: robottelo.hosts.Satellite instance
```
Comment (jyejare) [inline robottelo/hosts.py:272]:
```suggestion
        :param robottelo.hosts.Satellite satellite: robottelo.hosts.Satellite instance
```

OR

```suggestion
        :param satellite: robottelo.hosts.Satellite instance
```

File: tests/foreman/endtoend/test_api_endtoend.py [hunk 2]
```diff
@@ -1093,7 +1094,7 @@ def test_positive_end_to_end(self, fake_manifest_is_set):
             17. Create a new subnet
             18. Create a new domain
             19. Create a new hostgroup and associate previous entities to it
-            20. Provision a client
+            20. Provision a client  **  NOT CURRENTLY PROVISIONING
```
Comment (jyejare) [inline tests/foreman/endtoend/test_api_endtoend.py:1097]: Is string `** NOT CURRENTLY PROVISIONING` intentional ?

**Review by jyejare — APPROVED** [review]
ACK with non blocking comments.

File: tests/foreman/endtoend/test_api_endtoend.py (hunk 2 above)
Comment (mshriver) [reply tests/foreman/endtoend/test_api_endtoend.py:1097]: yep, the test doesn't actually provision through Satellite.

**Pull request merged**